src/styles/index.scss

```diff
@@ -1,3 +1,59 @@
+@import './utils/variables';
+
 body {
   margin: 0;
+  height: 100vh;
+  display: flex;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+.calendar {
+  display: flex;
+  margin: auto;
+  flex-wrap: wrap;
+  gap: $base-gap;
+  width: ($square-size * 7) + ($base-gap * 6);
+  padding: 10px;
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    height: $square-size;
+    width: $square-size;
+    background-color: $square-color;
+    border: 1px solid black;
+    transition: 500ms;
+
+    &::before {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      margin: auto;
+    }
+
+    &:hover {
+      background-color: $square-color-hover;
+      transform: translateY(-20px);
+    }
+  }
+}
+
+@for $day from 1 through 31 {
+  .calendar__day:nth-child(#{$day}):before {
+    content: '#{$day}';
+  }
+}
+
+@each $day, $index in $day-names {
+  .calendar--start-day-#{$day} .calendar__day:first-child {
+    margin-left: (($index - 1) * $square-size) + (($index - 1) * $base-gap);
```

Comment:
The calculation of the `margin-left` property is correct, but similar to the width of the calendar, it would be better to use a variable for the number of days in a week. This will make the code more readable and flexible. For example, you can use the `$days-in-week` variable here like so: `margin-left: (($index % $days-in-week) * ($square-size + $base-gap));`